### PR TITLE
Install dependencies using brew

### PR DESCRIPTION
**Pull request by mahrud**
This PR, which is on top of #1801, makes various improvements involving brew. It also adds a number of Dockerfiles that I've used for testing.

### CODE REVIEW

**Comment by mahrud** [comment]
@DanGrayson let's also merge this one for 1.17.1. Most of the changes are in [the docker directory](https://github.com/mahrud/M2/tree/feature/brew/M2/BUILD/docker).

**Comment by DanGrayson** [comment]
> @DanGrayson let's also merge this one for 1.17.1. Most of the changes are in [the docker directory](https://github.com/mahrud/M2/tree/feature/brew/M2/BUILD/docker).

okay

**Comment by mahrud** [comment]
I cancelled the last check because the changes are in readme files and dockerfiles. This can still be merged.

File: M2/GNUmakefile.in [hunk 1]
```diff
@@ -210,15 +210,7 @@ ifeq (@BUILD_givaro@,yes)
...
 
-all-in-Macaulay2:install-licenses
-
-install-licenses:all-in-submodules
-	: installing library and submodule licenses in @pre_licensesdir@
-	if [ -d $(BUILTLIBPATH)/licenses/. ] ; \
-	then rm -rf @pre_licensesdir@ ; \
-	     @INSTALL@ -d @pre_licensesdir@ ; \
-	     @TAR@ cf - -C $(BUILTLIBPATH)/licenses/. . | @TAR@ xf - -C @pre_licensesdir@ ; \
-	fi
```
Comment (DanGrayson) [inline M2/GNUmakefile.in:221]: Did the installation of licenses get moved somewhere else?

Comment (mahrud) [reply M2/GNUmakefile.in:221]: Yes, to libraries/Makefile.in

**Comment by mahrud** [comment]
So is this going to be in release-1.17 now?

**Comment by mahrud** [comment]
Oh I see, that branch is gone, nevermind.

**Pull request merged**